File: python/leetcode/merge.py
def merge1(nums1, m, nums2, n):
    """
    :type nums1: List[int]
    :type m: int
    :type nums2: List[int]
    :type n: int
    :rtype: None Do not return anything, modify nums1 in-place instead.
    
    将nums2加入到nums1 使用 nums[m:]
    再用sort()排序
    """
    nums1[m:] = nums2
    nums1.sort()
    return nums1



def merge2( nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        

     
nums1 = [1,2,3,0,0,0]
nums2 = [2,5,6]
m = 6
n= 3

nums = merge2(nums1,m,nums2,n)
print(nums)

File: java-leetcode/src/Merge.java
import java.util.Arrays;

public class Merge{
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int x = 0;
        for(int i = 0;i<nums1.length;i++){
            if(i>m&&x<=n){
                nums1[i]=nums2[x];
                x+=1;
            }
        }
        Arrays.sort(nums1);
    }
}